src/lazy/encoding.rs

```diff
@@ -118,8 +119,8 @@ pub struct BinaryEncoding_1_0;
 #[derive(Copy, Clone, Debug, Default)]
 pub struct BinaryEncoding_1_1;
 
-impl BinaryEncoding<'_> for BinaryEncoding_1_0 {}
-impl BinaryEncoding<'_> for BinaryEncoding_1_1 {}
+impl BinaryEncoding for BinaryEncoding_1_0 {}
+impl BinaryEncoding for BinaryEncoding_1_1 {}
```

Comment:
🪧 The lifetime in the `BinaryEncoding<'_>` and `TextEncoding<'_>` traits turned out to be unnecessary. 

This required using higher-ranked generics in the trait implementations, but keeping the lifetime out of the trait name simplified a lot of other changes.